packages/dds/map/src/mapKernel.ts

```diff
@@ -641,6 +956,29 @@ export class MapKernel {
 		return !local;
 	}
 
+	private ackPendingSetOp(op: IMapKeyOperation, pendingMessageId?: number): void {
+		// If the message id of the "earliest" existing pending op matches the current pendingMessageId,
+		// we need to ack this op
+		const pendingSetIds = this.pendingSetTracker.get(op.key);
+		if (pendingSetIds !== undefined && pendingSetIds[0] === pendingMessageId) {
+			if (!this.ackedKeysIndexTracker.has(op.key)) {
+				this.ackedKeysIndexTracker.set(op.key, ++this.creationIndex);
```

Comment:
Always good to give an idea of when that's not true in comments like these. I believe this is the logic:


```suggestion
		// If the message id of the "earliest" existing pending op matches the current pendingMessageId,
		// we need to ack this op. This is true iff there are no pending deletes.
```